Add encode/decode size calculation

jagerman opened the pull request.

jagerman committed:
Add encode/decode size calculation

majestrate commented on oxenmq/hex.h:
```diff
@@ -124,6 +124,11 @@ template <typename CharT>
+/// Returns the number of characters required to encode a hex string from the given number of bytes.
+inline constexpr size_t to_hex_size(size_t byte_size) { return byte_size * 2; }
+/// Returns the number of bytes required to decode a hex string of the given size.
+inline constexpr size_t from_hex_size(size_t hex_size) { return hex_size / 2; }
```
this should handle when `hex_size` is odd

jagerman replied:
Hex size being odd doesn't pass `is_hex()`, and invoked UB on `from_hex` if you use it.

jagerman committed:
Add `(valid)` to descriptions

E.g. passing it an odd number for from_hex_size() isn't a valid hex
string (it fails `is_hex()`).

jagerman commented:
Closing because I included this in #52, but ended up having to move it so it conflicts.
